merlin/models/tf/blocks/core/transformations.py

```diff
@@ -50,20 +56,44 @@ def compute_output_shape(self, input_shape):
 @Block.registry.register("as-dense")
 @tf.keras.utils.register_keras_serializable(package="merlin.models")
 class AsDenseFeatures(TabularBlock):
+    def __init__(self, max_seq_length: Optional[int] = None, **kwargs):
+        super().__init__(**kwargs)
+        self.max_seq_length = max_seq_length
+
     def call(self, inputs: TabularData, **kwargs) -> TabularData:
         outputs = {}
         for name, val in inputs.items():
             if isinstance(val, tuple):
                 values = val[0][:, 0]
                 row_lengths = val[1][:, 0]
-                outputs[name] = tf.RaggedTensor.from_row_lengths(values, row_lengths).to_tensor()
+                ragged = tf.RaggedTensor.from_row_lengths(values, row_lengths)
+                if self.max_seq_length:
+                    outputs[name] = ragged.to_tensor(shape=[None, self.max_seq_length])
+                else:
+                    outputs[name] = tf.squeeze(ragged.to_tensor())
```

Comment:
Should we raise an exception if `max_seq_length` is not defined for a sparse feature?

Reply:
I think it’s fine without since we can still convert it, then the max-lenght within the batch will be used as the max which is OK for certain scenarios I think. Does that make sense?

Reply:
Ok. It makes sense. And the users can always reshape the tensors on their side too